File: DSRFQ.Web/Modules/Master/Settings/SettingsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace DSRFQ.Master.Columns;

[ColumnsScript("Master.Settings")]
[BasedOnRow(typeof(SettingsRow), CheckNames = true)]
public class SettingsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink, Width(180)]
    public bool DatumAddMissing { get; set; }
    [Width(180)]
    public decimal DatumAddMinConfidence { get; set; }
    [Width(160)]
    public string SubNumberSeparator { get; set; }
    [DateTimeFormatter(DisplayFormat = "yyyy-MM-dd HH:mm:ss"), Width(140)]
    public DateTime UpdateDate { get; set; }
}

File: DSRFQ.Web/Modules/ServerTypes/Master/SettingsColumns.ts
import { fieldsProxy } from "@serenity-is/corelib";
import { SettingsRow } from "./SettingsRow";

export interface SettingsColumns {
    Id: SettingsRow;
    DatumAddMissing: SettingsRow;
    DatumAddMinConfidence: SettingsRow;
    SubNumberSeparator: SettingsRow;
    UpdateDate: SettingsRow;
}

export class SettingsColumns {
    static readonly columnsKey = 'Master.Settings';
    static readonly Fields = fieldsProxy<SettingsColumns>();
}
